ui/components/app/transaction-decoding/transaction-decoding.component.js

```diff
@@ -0,0 +1,249 @@
+import React, { useContext, useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Spinner from '../../ui/spinner';
+import ErrorMessage from '../../ui/error-message';
+import fetchWithCache from '../../../helpers/utils/fetch-with-cache';
+import { useSelector } from 'react-redux';
+import * as Codec from '@truffle/codec';
+import { forAddress } from '@truffle/decoder';
+import inspect from 'browser-util-inspect';
+import { getSelectedAccount, getCurrentChainId } from '../../../selectors';
+import {
+  FETCH_PROJECT_INFO_URI,
+  FETCH_SUPPORTED_NETWORKS_URI,
+} from './constants';
+import { hexToDecimal } from '../../../helpers/utils/conversions.util';
+import { I18nContext } from '../../../contexts/i18n';
+import { transformTxDecoding } from './transaction-decoding.util';
+import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+
+import Address from './components/decoding/address';
+import CopyRawData from './components/ui/copy-raw-data/';
+
+export default function TransactionDecoding({ to = '', inputData: data = '' }) {
+  const t = useContext(I18nContext);
+  const [tx, setTx] = useState([]);
+  const { address: from } = useSelector(getSelectedAccount);
+  const network = hexToDecimal(useSelector(getCurrentChainId));
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+      try {
+        const networks = await fetchWithCache(
```

Comment:
What is this fails due to network or server failure? Do the error messages below cover it?

Reply:
It looks like fetch-with-cache handles error appropriately ! so we have reverted all redundant error throwing !